exporter/sentryexporter/transport.go

```diff
@@ -0,0 +1,96 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sentryexporter
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+)
+
+const defaultTimeout = time.Second * 30
+
+// A SentryTransport is used to deliver events to a remote server
+type SentryTransport struct {
+	DSN       *sentry.Dsn
+	client    *http.Client
+	transport http.RoundTripper
+
+	// HTTP Client request timeout. Defaults to 30 seconds.
+	Timeout time.Duration
+}
+
+// NewSentryTransport returns a new pre-configured instance of SentryTransport
+func NewSentryTransport() *SentryTransport {
+	return &SentryTransport{
+		Timeout: defaultTimeout,
+	}
+}
+
+// Configure configures the SentryTransport based on provided config
+func (t *SentryTransport) Configure(config *Config) error {
+	DSN, err := sentry.NewDsn(config.DSN)
+	if err != nil {
+		return err
+	}
+
+	t.DSN = DSN
+
+	t.client = &http.Client{
+		Transport: t.transport,
+		Timeout:   t.Timeout,
+	}
+
+	return nil
+}
+
+// SendTransaction send a transaction to a remote server
+func (t *SentryTransport) SendTransaction(transaction *SentryTransaction) error {
+	body, err := json.Marshal(transaction)
+	if err != nil {
+		return err
+	}
+
+	request, _ := http.NewRequest(
+		http.MethodPost,
+		t.DSN.StoreAPIURL().String(),
+		bytes.NewBuffer(body),
```

Comment:
Can we `gzip` compress the payload instead?

`sentry-go` is probably doing it? (/cc @rhcarvalho )

```suggestion
		bytes.NewBuffer(body),
```

Reply:
I'll add gzip here and then port these changes to `sentry-go`